lib/elixir_analyzer/exercise_test/common_checks/indentation.ex

```diff
@@ -0,0 +1,27 @@
+defmodule ElixirAnalyzer.ExerciseTest.CommonChecks.Indentation do
+  @moduledoc """
+  Reports if tabs were used for indentation.
+  """
+
+  alias ElixirAnalyzer.Constants
+  alias ElixirAnalyzer.Comment
+
+  @spec run(Macro.t(), String.t()) :: [{:pass | :fail | :skip, %Comment{}}]
+  def run(ast, string) do
+    if uses_tabs_for_indentation?(ast, string) do
+      [
+        {:fail,
+         %Comment{
+           type: :informative,
+           comment: Constants.solution_indentation()
+         }}
+      ]
+    else
+      []
+    end
+  end
+
+  defp uses_tabs_for_indentation?(ast, string) do
+    String.contains?(string, "\t") && !String.contains?(Macro.to_string(ast), "\\t")
```

Comment:
What if tabs are used for indentation AND is used in strings?

Reply:
Maybe you could count the tabs in `string` and that should match the numbers of tabs in `Macro.to_string(ast)`.

Reply:
I think this is a good point, what if a tab appears in a string literal which is intended to be there and is not being used for indentation, will this give a false-positive error?

Maybe a safer approach is:
```elixir
tab_in_leading_whitespace = ~r/^\s*\t/

string
|> String.split("\n")
|> Enum.any(&(String.match?(&1, tab_in_leading_whitespace)))
```

Reply:
This will trigger a fail when a multiline string starts with tabs, I think.

Reply:
Ok, then more strictly:

```elixir
tab_in_leading_whitespace = ~r/^\t/
```

Reply:
This would miss `\s\t\s` which is arguably a way worse indentation style :D
I think counting the number of "intended" tabs and making sure that that is the only tabs in the source would work well.

Reply:
It would also miss `\s\s\t\s` -- my point was to remove the false-positive case.  If you have an implementation of your idea to test that would be good.

Reply:
OK I did: [here](https://github.com/jiegillet/elixir-analyzer/commit/90182774492c4932f60c5aa0245ffe72b717e5f0) and [here](https://github.com/jiegillet/elixir-analyzer/commit/7c6156caccbcf3c009c1f596108d3d4470225bec). 
(sorry my GitHub-Foo is a bit weak, there is probably a better way to link to these)

Reply:
I took over @jiegillet's test, solution, and added a bunch more tests for more corner cases. Thank  you!

Reply:
Neat, tabs and escaped tabs into multiline strings became very strange to me after a while, the solution was obvious: add more tests :)